reana_client/cli/workflow.py

```diff
@@ -738,6 +741,17 @@ def workflow_logs(ctx, workflow, access_token, json_format, steps=None):  # noqa
                 steps = list(set(steps))
             response = get_workflow_logs(workflow, access_token, steps)
             workflow_logs = json.loads(response["logs"])
+
+            if failed:
+                finished_steps = [
+                    key
+                    for key, value in workflow_logs["job_logs"].items()
+                    if value["status"] != "failed"
+                ]
+
+                for job_id in finished_steps:
+                    del workflow_logs["job_logs"][job_id]
```

Comment:
All works fine except this message, which can be misleading:
```console
$ reana-client logs -w workflow.1 --step success --failed 
The logs of step(s) success were not found, check for spelling mistakes in the step names.
```
This happens when none of the steps meets the "failed" condition. I think it would make sense to check if the `workflow_logs["job_logs"]` is empty here and return a more accurate message, what do you think?